### PR TITLE
SOLR-17149: Fix backup/restore for large collections.

### PR DESCRIPTION
**Pull request by psalagnac**
https://issues.apache.org/jira/browse/SOLR-17149

# Description

A regression was introduced with [SOLR-16879](https://issues.apache.org/jira/browse/SOLR-16879) that makes backup/restore failing when there are more than 5 shards per node.

# Solution

Replace the bounded queue for async core admin tasks by an unbounded one.

# Tests

No unit tests added.
Manually checked we don't have any tasks rejected by the executor.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/core/CoreContainer.java [hunk 1]
```diff
@@ -412,7 +412,8 @@ public CoreContainer(NodeConfig config, CoresLocator locator, boolean asyncSolrC
         new OrderedExecutor(
             cfg.getReplayUpdatesThreads(),
             ExecutorUtil.newMDCAwareCachedThreadPool(
```
[the remark this answers is not in the document]
Comment (gerlowskija) [reply solr/core/src/java/org/apache/solr/core/CoreContainer.java:414]: > I may remove the capacity parameter and always use Integer.MAX_VALUE for the queue capacity.

I suspect you're right that this was an oversight or mistake.  But I'd file a separate ticket for changing this if you're interested.  It's likely a safe change, but it seems like something we'd want to "bake" for longer before it goes into a release.

**Comment by gerlowskija** [comment]
Hi @psalagnac - this LGTM, generally.

The one thing missing IMO is some modifications to `AbstractIncrementalBackupTest` to cover the many-shards case.  I took a stab at these locally and have code ready to share, put it looks like your branch isn't configured to allow others to push to it:

```
$ git push psalagnac SOLR-17149:SOLR-17149
Enumerating objects: 84, done.
Counting objects: 100% (84/84), done.
Delta compression using up to 28 threads
Compressing objects: 100% (32/32), done.
Writing objects: 100% (51/51), 13.84 KiB | 4.61 MiB/s, done.
Total 51 (delta 22), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (22/22), completed with 19 local objects.
To github.com:psalagnac/solr.git
 ! [remote rejected]         SOLR-17149 -> SOLR-17149 (**permission denied**)
error: failed to push some refs to 'github.com:psalagnac/solr.git'
```


If you're willing to accept my test changes, could you take a look at your branch settings and let me know when they're updated?  Or alternatively, if you'd rather make the changes yourself, that's fine too.

File: solr/core/src/java/org/apache/solr/core/CoreContainer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (psalagnac) [reply solr/core/src/java/org/apache/solr/core/CoreContainer.java:414]: Thanks, makes sense.
I'll fill another ticket for that later so this is not in 9.5.

**Comment by psalagnac** [comment]
@gerlowskija Not sure why you got denied. I already checked the _"Allow edits and access to secrets by maintainers"_ box.

Can this be because of branch name case? (`SOLR-17149` vs `solr-17149`)

**Comment by gerlowskija** [comment]
Gah - that must've been it!  Thanks for catching my mistake.  I've pushed my test-changes, and run 'test'+'check' locally.  So I'll merge and backport.  Thanks for the quick fix @psalagnac !

**Comment by psalagnac** [comment]
Thanks for the test. Looks good to me.

**Pull request merged**